### PR TITLE
Dit 81 ruling pdf

### PR DESCRIPTION
**Pull request by SteveSugden**
Adds endpoint to generate Ruling PDF for a case that has a decision.  
Updates the links on the Ruling tab on the trader home page.
Some code tidy (fix warnings) in recently merged files

### CODE REVIEW

File: app/controllers/PdfDownloadController.scala [hunk 1]
```diff
@@ -42,10 +42,20 @@ class PdfDownloadController @Inject()(appConfig: FrontendAppConfig,
         pdfService.generatePdf(applicationPdf(appConfig, c, attachmentData)).map { binaryFile =>
           Results.Ok(binaryFile.content)
             .as(binaryFile.contentType)
-            .withHeaders(CONTENT_DISPOSITION -> s"filename=confirmation_$reference.pdf")
+            .withHeaders(CONTENT_DISPOSITION -> s"filename=BTIConfirmation$reference.pdf")
         }
       }
     }
   }
```
Comment (gokyo) [inline app/controllers/PdfDownloadController.scala]: The two methods in this class are very similar.
You could create a common private method used by both (DRY principle)

File: app/models/Case.scala [hunk 2]
```diff
@@ -35,11 +35,12 @@ case class Case
...
-  def hasEoriNumber(eoriNumber: String): Boolean = {
-    application.holder.eori == eoriNumber || application.agent.exists(_.eoriDetails.eori == eoriNumber)
-  }
+  def hasEoriNumber(eoriNumber: String): Boolean = application.holder.eori == eoriNumber || application.agent.exists(_.eoriDetails.eori == eoriNumber)
 
   def hasActiveDecision: Boolean = this.decision.flatMap(_.effectiveEndDate).exists(_.compareTo(Instant.now) >= 0)
 
   def hasExpiredDecision: Boolean = this.decision.flatMap(_.effectiveEndDate).exists(_.compareTo(Instant.now) < 0)
+
+  def hasRuling: Boolean = decision.exists(d => d.effectiveStartDate.isDefined && d.effectiveEndDate.isDefined)
```
Comment (gokyo) [inline app/models/Case.scala]: is this the proper way to make sure there is a ruling ?

File: app/controllers/PdfDownloadController.scala (hunk 1 above)
Comment (SteveSugden) [reply app/controllers/PdfDownloadController.scala]: Pulled out the common code into a private method

File: app/models/Case.scala (hunk 2 above)
Comment (SteveSugden) [reply app/models/Case.scala]: Not sure if that is the "complete" definition of "case has a ruling".    It's good enough (IMO) to ensure that a ruling cert pdf will have all of the information it requires (i.e. a Decision block with start and end dates).

The only other additional check I can think of is that the case status is Completed or Canceled.  I'll check with Matt/Joe

Comment (pllaurado) [reply app/models/Case.scala]: Those are my acceptance criteria in DIT-79 about what to show as a Ruling : 

> Ruling status mappings must be presented as follows:
> "Active" where Case.Status = Complete AND Current Date <= ruling End Date
> "Expired" where Case.Status = Complete and Current Date > ruling End Date
> "Cancelled" where Case.Status = Cancelled

Comment (SteveSugden) [reply app/models/Case.scala]: Thanks - I've added a question on my ticket - https://jira.tools.tax.service.gov.uk/browse/DIT-81

File: app/controllers/PdfDownloadController.scala [hunk 3]
```diff
@@ -42,10 +42,20 @@ class PdfDownloadController @Inject()(appConfig: FrontendAppConfig,
         pdfService.generatePdf(applicationPdf(appConfig, c, attachmentData)).map { binaryFile =>
           Results.Ok(binaryFile.content)
             .as(binaryFile.contentType)
-            .withHeaders(CONTENT_DISPOSITION -> s"filename=confirmation_$reference.pdf")
+            .withHeaders(CONTENT_DISPOSITION -> s"filename=BTIConfirmation$reference.pdf")
```
Comment (pllaurado) [inline app/controllers/PdfDownloadController.scala]: I don't which are the standards / conventions  for file name. Is this name ok ? BTIConfirmation$reference.pdf

Comment (SteveSugden) [reply app/controllers/PdfDownloadController.scala]: I checked on the file names with Matt yesterday.   He preferred / was happy with the new format

File: app/models/Case.scala (hunk 2 above)
Comment (gokyo) [reply app/models/Case.scala]: @SteveSugden 
as agreed with @pllaurado , we would like this method to assert the case status (you can keep the dates assertion, but perhaps better to get rid of it and just check the case status == completed / decision_made)

Comment (SteveSugden) [reply app/models/Case.scala]: Ok.  But it will have to be status = Completed or Cancelled

**Pull request merged**